src/xrGame/agent_enemy_manager.cpp

```diff
@@ -664,31 +634,24 @@ ALife::_OBJECT_ID CAgentEnemyManager::wounded_processor(const CEntityAlive* obje
     return (ALife::_OBJECT_ID(-1));
 }
 
-class find_wounded_predicate
-{
-private:
-    const CEntityAlive* m_object;
-
-public:
-    IC find_wounded_predicate(const CEntityAlive* object)
-    {
-        m_object = object;
-        VERIFY(m_object);
-    }
-
-    IC bool operator()(const CAgentEnemyManager::WOUNDED_ENEMY& enemy) const { return (enemy.first == m_object); }
-};
-
 void CAgentEnemyManager::wounded_processor(const CEntityAlive* object, const ALife::_OBJECT_ID& wounded_processor_id)
 {
-    VERIFY(std::find_if(m_wounded.begin(), m_wounded.end(), find_wounded_predicate(object)) == m_wounded.end());
+    VERIFY(std::find_if(m_wounded.begin(), m_wounded.end(), [object](const WOUNDED_ENEMY& enemy)
+    {
+        VERIFY(object);
```

Comment:
Looks like this check will be executed every time the predicate is called, isn't it?
Original code did that check only one time in predicate constructor.